igvc_perception/src/pointcloud_filter/fast_segment_filter/fast_segment_filter.cpp

```diff
@@ -0,0 +1,317 @@
+#include <pointcloud_filter/fast_segment_filter/fast_segment_filter.h>
+
+namespace pointcloud_filter
+{
+void FastSegmentFilter::filter(pointcloud_filter::Bundle &bundle)
+{
+  segments_.clear();
+  ground_points_.clear();
+  nonground_points_.clear();
+
+  for (const auto &point : bundle.pointcloud->points)
+  {
+    int segment_id = getAngleFromPoint(point) * config_.num_segments / 360;
+    segments_[segment_id].raw_points_.emplace_back(point);
+  }
+  processSegments();
+  getLinesFromSegments();
+  if (config_.debug_viz)
+  {
+    debugViz();
+  }
+  ground_points_.header.frame_id = "/lidar";
+  nonground_points_.header.frame_id = "/lidar";
+  classifyPoints(ground_points_, nonground_points_);
+  ground_pub_.publish(ground_points_);
+  nonground_pub_.publish(nonground_points_);
+
+  bundle.free_pointcloud->points = std::move(ground_points_.points);
+  bundle.occupied_pointcloud->points = std::move(nonground_points_.points);
+}
+
+double Line::distFromPoint(const Prototype point)
+{
+  Eigen::Vector3d vect;
+  vect(0) = point.point_.x;
+  vect(1) = point.point_.y;
+  vect(2) = point.point_.z;
+  double t = -(intercept_ - vect).dot(params_) / (params_.dot(params_));
```

Comment:
Maybe use `params_.normSquared()` instead of `params_.dot(params_)` for clarity? Also, I think `t` can be clarified a bit, as it takes a second to see what's happening.